rest-service/manager_rest/cluster_status_manager.py

```diff
@@ -236,511 +195,262 @@ def get_concurrent_status_checker():
-        services, services_not_available = _services_from_prometheus_response(
-            prometheus_reponse, _metrics_for_host, node_tuples)
-        # If the remote metrics are not available, assume service_node is down
-        # and combine not_available_nodes with those available.
-        services.update(services_not_available)
-        all_services[ip_address] = services[ip_address]
-    return all_services
-
-
-def _services_from_prometheus_response(prometheus_response,
-                                       select_node_metrics_func,
-                                       node_tuples):
-    # node_tuples is a list of two-element tuples: (private_ip, service_type)
-    services = {}
-    services_not_available = []
-    for node_tuple in node_tuples:
-        node_metrics = select_node_metrics_func(prometheus_response,
-                                                node_tuple[0])
-        if node_metrics:
-            services[node_tuple[0]] = _service_statuses(node_metrics)
-        else:
-            services_not_available.append(node_tuple)
-    return services, services_not_available
-
-
-def _service_statuses(node_metrics):
-    reported_services = {}
-    for node_metric in node_metrics:
-        service_id = node_metric.get('metric', {}).get('name')
-        service = _get_cloudify_service_description(service_id)
-        if not service:
-            continue
-        process_manager = _get_process_manager(node_metric.get('metric'))
-        service_status = _get_service_status(
-            service_id, service, process_manager,
-            node_metric.get('value')[1] == '1')
-        if reported_services.get(service.name, {}).get('extra_info', {}).get(
-                process_manager, {}).get('instances'):
-            reported_services[service.name]['extra_info'][process_manager][
-                'instances'].append(service_status['extra_info'][
-                                        process_manager]['instances'])
-        else:
-            reported_services[service.name] = service_status
-    return reported_services
+    for metric in node['metrics']:
+        if not metric['healthy']:
+            return NodeServiceStatus.INACTIVE
 
+    return NodeServiceStatus.ACTIVE
 
-def _get_process_manager(metric):
-    if 'systemd' in metric.get('__name__'):
-        return 'systemd'
-    elif 'supervisord' in metric.get('__name__'):
-        return 'supervisord'
-    else:
-        return 'unknown'
 
+def _parse_prometheus_results(prometheus_results):
+    service_results = []
+    metric_results = {}
 
-def _get_cloudify_service_description(metric_name):
-    if metric_name in SERVICE_DESCRIPTIONS:
-        return SERVICE_DESCRIPTIONS[metric_name]
-    elif metric_name.endswith('.service'):
-        return _get_cloudify_service_description(metric_name[:-8])
-    return None
+    for result in prometheus_results:
+        metric = result.get('metric', {})
+        value = result.get('value', [0, ''])
```

Comment:
maybe `timestamp, is_healthy = ...` unpack here so that those things have a name? with just `[0, '']`, it's hard to tell what are we looking at

Reply:
Makes sense, modified.